source/alloc/src/heap.rs

```diff
@@ -1,505 +1,128 @@
-                // Can we get an exclusive heap handle?
-                if let Ok(mut hg) = self.lock() {
-                    match hg.alloc_fixed_vec(capacity) {
-                        Ok(hb) => {
-                            // Yes! Return our allocated item
-                            return hb;
-                        }
-                        Err(_) => {
-                            // Nope, the allocation failed.
-                        }
-                    }
-                }
-                // We weren't inhibited before, but something failed. Inhibit
-                // further allocations to prevent starving waiting allocations
-                self.inhibit_alloc.store(true, Ordering::Release);
-            }
-
-            // Didn't succeed, wait until we've done some de-allocations
-            self.heap_wait.wait().await.unwrap();
-        }
+    unsafe fn alloc(&self, layout: core::alloc::Layout) -> *mut u8 {
+        <std::alloc::System as GlobalAlloc>::alloc(self, layout)
     }
 
-    pub async fn allocate_raw(&'static self, layout: Layout) -> NonNull<()> {
-        loop {
-            // Is the heap inhibited?
-            if !self.inhibit_alloc.load(Ordering::Acquire) {
-                // Can we get an exclusive heap handle?
-                if let Ok(mut hg) = self.lock() {
-                    match hg.alloc_raw(layout) {
-                        Ok(hb) => {
-                            // Yes! Return our allocated item
-                            return hb;
-                        }
-                        Err(_) => {
-                            // Nope, the allocation failed.
-                        }
-                    }
-                }
-                // We weren't inhibited before, but something failed. Inhibit
-                // further allocations to prevent starving waiting allocations
-                self.inhibit_alloc.store(true, Ordering::Release);
-            }
-
-            // Didn't succeed, wait until we've done some de-allocations
-            self.heap_wait.wait().await.unwrap();
-        }
+    unsafe fn dealloc(&self, ptr: *mut u8, layout: core::alloc::Layout) {
+        <std::alloc::System as GlobalAlloc>::dealloc(self, ptr, layout)
     }
 }
 
-/// Deallocate an unsized allocation with the provided `Layout`.
-///
-/// # Safety
-///
-/// - `ptr` *must* have been returned by a call to [`AHeap::allocate_raw`] or
-///   [`HeapGuard::alloc_raw`]!
-/// - `layout` *must* be the same `Layout` that was provided to the original
-///   call to [`AHeap::allocate_raw`] or[`HeapGuard::alloc_raw`]!
-pub unsafe fn deallocate_raw(ptr: NonNull<()>, layout: Layout) {
-    let ptr = ActiveUnsized::from_raw(ptr, layout);
-    ActiveUnsized::yeet(ptr, layout);
-}
-
-/// A guard type that provides mutually exclusive access to the allocator as
-/// long as the guard is held.
-pub struct HeapGuard {
-    aheap: &'static AHeap,
+pub struct AHeap2<U> {
```

Comment:
nitpicky, sorry: since we deleted the original `AHeap` type, can't this just be `AHeap`, rather than `AHeap2`?